In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('../data/AdSmartABdata.csv')
df.head()

auction_id experiment        date  hour  \
0  0008ef63-77a7-448b-bd1e-075f42c55e39    exposed  2020-07-10     8   
1  000eabc5-17ce-4137-8efe-44734d914446    exposed  2020-07-07    10   
2  0016d14a-ae18-4a02-a204-6ba53b52f2ed    exposed  2020-07-05     2   
3  00187412-2932-4542-a8ef-3633901c98d9    control  2020-07-03    15   
4  001a7785-d3fe-4e11-a344-c8735acacc2c    control  2020-07-03    15   

          device_make  platform_os                browser  yes  no  
0  Generic Smartphone            6          Chrome Mobile    0   0  
1  Generic Smartphone            6          Chrome Mobile    0   0  
2               E5823            6  Chrome Mobile WebView    0   1  
3   Samsung SM-A705FN            6               Facebook    0   0  
4  Generic Smartphone            6          Chrome Mobile    0   0

In [4]:
#Determining the users who responded to the ads

response_df = df.query("(yes == 1 or no == 1)")
response_df.shape[0]

1243

In [5]:
#Total number of users
df.shape[0]

8077

Out of the 8077 users, only 1243 responded to the questionnaire

In [10]:
#Getting clean data containing only users who responded alongside relevant columns for testing

clean_df = response_df[['auction_id', 'experiment', 'yes', 'no']]
clean_df

auction_id experiment  yes  no
2     0016d14a-ae18-4a02-a204-6ba53b52f2ed    exposed    0   1
16    008aafdf-deef-4482-8fec-d98e3da054da    exposed    1   0
20    00a1384a-5118-4d1b-925b-6cdada50318d    exposed    0   1
23    00b6fadb-10bd-49e3-a778-290da82f7a8d    control    1   0
27    00ebf4a8-060f-4b99-93ac-c62724399483    control    0   1
...                                    ...        ...  ...  ..
8059  ffa08ff9-a132-4051-aef5-01a9c79367bc    exposed    1   0
8063  ffb176df-ecd2-45d3-b05f-05b173a093a7    exposed    1   0
8064  ffb79718-6f25-4896-b6b3-e58b80a6e147    control    0   1
8069  ffca1153-c182-4f32-9e90-2a6008417497    control    0   1
8071  ffdfdc09-48c7-4bfb-80f8-ec1eb633602b    exposed    0   1

[1243 rows x 4 columns]

In [ ]:
aware_users